### PR TITLE
Support api versioning for subscriptions

### PR DESCRIPTION
**Pull request by justin-tay**
## Description
Adds support for GraphQL subscriptions to handle API versioning.

## Motivation and Context
GraphQL subscriptions couldn't easily handle API versioning.

## How Has This Been Tested?
Added the appropriated tests and manually with modified example applications.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-datastore/elide-datastore-jms/src/main/java/com/yahoo/elide/datastores/jms/websocket/SubscriptionWebSocketConfigurator.java [hunk 1]
```diff
@@ -81,22 +89,68 @@ public class SubscriptionWebSocketConfigurator extends ServerEndpointConfig.Conf
...
+        Object httpSession = request.getHttpSession();
+        if (httpSession != null)
+        {
+            userProperties.put("session", httpSession);
+        }
+
+        Principal userPrincipal = request.getUserPrincipal();
+        if (userPrincipal != null)
+        {
+            userProperties.put("userPrincipal", userPrincipal);
```
Comment (aklish) [inline elide-datastore/elide-datastore-jms/src/main/java/com/yahoo/elide/datastores/jms/websocket/SubscriptionWebSocketConfigurator.java:150]: Is there some plan later to potentially use this principal to construct the Elide user object?

File: elide-graphql/src/main/java/com/yahoo/elide/graphql/subscriptions/websocket/SubscriptionWebSocket.java [hunk 2]
```diff
@@ -231,21 +258,52 @@ private SessionHandler findSession(Session wrappedSession) {
...
+        Map<String, List<String>> headers = new LinkedHashMap<>(session.getRequestParameterMap());
+        // Get the properties from the handshake set in SubscriptionWebSocketConfigurator
+        if (session.getUserProperties().get("headers") instanceof Map handshakeHeaders) {
+            headers.putAll(handshakeHeaders);
+        }
+
+        Route route = routeResolver.resolve(MediaType.APPLICATION_JSON, baseUrl, path, headers,
+                session.getRequestParameterMap());
+
+        String apiVersion = route.getApiVersion();
```
Comment (aklish) [inline elide-graphql/src/main/java/com/yahoo/elide/graphql/subscriptions/websocket/SubscriptionWebSocket.java:284]: The route resolver here that resolved the apiVersion from parameters looks at a different parameter name than the old code.  The old code looked for ApiVersion in the parameters and the new code looks for "v" in the ParameterRouteResolver.

**Review by aklish — CHANGES_REQUESTED** [review]

File: elide-datastore/elide-datastore-jms/src/main/java/com/yahoo/elide/datastores/jms/websocket/SubscriptionWebSocketConfigurator.java (hunk 1 above)
Comment (justin-tay) [reply elide-datastore/elide-datastore-jms/src/main/java/com/yahoo/elide/datastores/jms/websocket/SubscriptionWebSocketConfigurator.java:150]: This user principal here should be the same one that can be accessed from the `jakarta.websocket.Session.getUserPrincipal` method.

The default `SubscriptionWebSocket.UserFactory` already creates a user using `session -> new User(session.getUserPrincipal());`

File: elide-graphql/src/main/java/com/yahoo/elide/graphql/subscriptions/websocket/SubscriptionWebSocket.java (hunk 2 above)
Comment (justin-tay) [reply elide-graphql/src/main/java/com/yahoo/elide/graphql/subscriptions/websocket/SubscriptionWebSocket.java:284]: The `ApiVersion` that it looked for previously was more in line with what was looked for from the header. In the new code this is catered by initializing the header values from the request parameter map in the `createSessionHandler` using `Map<String, List<String>> headers = new LinkedHashMap<>(session.getRequestParameterMap());` in addition to the other values found as HTTP headers. So the `HeaderRouteResolver` should resolve the API Version if configured with `ApiVersion`.

**Review by aklish — APPROVED** [review]

**Pull request merged**